packages/story-editor/src/components/fontCheck/fontCheckDialog.js

```diff
@@ -0,0 +1,196 @@
+  color: black;
+  align-self: flex-start;
+  align-items: center;
+  font-weight: normal;
+  margin-right: 40px;
+`;
+
+const defaultFont = {
+  font: {
+    id: TEXT_ELEMENT_DEFAULT_FONT?.family,
+    ...TEXT_ELEMENT_DEFAULT_FONT,
+  },
+};
+
+export const FontCheckDialog = ({ isOpen, onClose, missingFont }) => {
+  const { dashboardLink, dashboardSettingsLink, isRTL } = useConfig();
+  const [suggestedFont, setSuggestedFont] = useState(defaultFont);
+  const { updateElementsByFontFamily } = useStory((state) => ({
+    updateElementsByFontFamily: state.actions.updateElementsByFontFamily,
+  }));
+
+  /**
+   * Handles updating suggested font which is passed to the FontPicker
+   *
+   * @param {Object} [evt] Font object.
+   * @return void
+   */
+  const onPushUpdate = useCallback((evt) => {
+    if (evt?.font) {
+      setSuggestedFont({
+        font: { id: evt.font.family, family: evt.font.family },
+      });
+    }
+  }, []);
+
+  const onSettingsClick = useCallback((evt) => {
+    trackClick(evt, 'click_video_optimization_settings');
+  }, []);
+
+  const updateMissingFontWithDefault = useCallback(() => {
+    const { id, ...font } = defaultFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+  }, [onClose, missingFont, updateElementsByFontFamily]);
+
+  const updateMissingFontWithSelected = useCallback(() => {
+    const { id, ...font } = suggestedFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+  }, [suggestedFont, onClose, missingFont, updateElementsByFontFamily]);
+
+  const Chevron = isRTL ? ChevronRight : ChevronLeft;
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      title={__('Missing Fonts', 'web-stories')}
+      contentLabel={__('Missing Fonts', 'web-stories')}
+      actions={
+        <>
+          <Link href={dashboardLink}>
+            <DashboardLink>
+              <Chevron />
+              {__('Back to dashboard', 'web-stories')}
+            </DashboardLink>
+          </Link>
+          <Button
+            type={BUTTON_TYPES.SECONDARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithDefault}
+          >
+            {__('Open anyway', 'web-stories')}
+          </Button>
+          <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithSelected}
+          >
+            {__('Replace font', 'web-stories')}
+          </Button>
+        </>
+      }
+    >
+      <DialogContent>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__(
+            'This story contains the following font that is no longer available:',
+            'web-stories'
+          )}
+        </Text>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL} isBold>
+          {missingFont}
+        </Text>
```

Comment:
@swissspidy Should this be one string. Like this.

```
sprintf( 'This story contains the following font that is no longer available: %s', missingFont )
```

Reply:
This is fine because it's not really used for string concatenation. It's more like a heading plus a list of items. So this is OK.

In any case, the wording will most likely be tweaked in the future anyway.